### DISCUSSION TITLE
Option to pass in client instead of clientPromise to MongoDB Adapter

### DISCUSSION THREAD

**Discussion by AnishDe12020** [nextauthjs/next-auth#3960] [Ideas]
### Title

Option to pass in client instead of clientPromise to MongoDB Adapter

### How to reproduce ☕️

Not a bug report.

### Your question/bug report 📓

Currently, the MongoDB Adapter takes in a `clientPromise` which is awaited on the adapter and a connection is established. However, if I want to use MongoDB in my application for anything other than auth, I need to open another connection to the database. It would be helpful if we could (or at least had the option to) pass in a `MongoClient` object to the adapter. 

### Contributing  🙌🏽

Yes, I am

**Comment by balazsorban44** [comment]
This is taken from the recommendations in the Next.js example repository: https://github.com/vercel/next.js/blob/canary/examples/with-mongodb/lib/mongodb.js

This should actually hold the connection if you use it in other places compared to what you are suggesting, where the adapter would need to create its own connection.
